service/key/key.go

```diff
@@ -263,19 +265,34 @@ func RemoteWritePasswordKey() string {
 	return "password"
 }
 
-// IsCAPICluster returns true if the cluster is in v1alpha3 and does not have the "azure-operator.giantswarm.io/version" label added by the azure operator.
-// We do not have a provider agnostic label like "giantswarm.io/version" to define this.
+// IsCAPICluster returns true if the cluster has any of the legacy labels such as azure-operator.giantswarm.io/version.
 func IsCAPICluster(obj metav1.Object) bool {
 	// TODO once we have migrated all clusters to CAPI, we can remove this
-	switch v := obj.(type) {
-	case *capiv1alpha3.Cluster:
-		if _, ok := v.Labels["azure-operator.giantswarm.io/version"]; ok {
+
+	checker := func(labels map[string]string) bool {
+		if _, ok := labels["azure-operator.giantswarm.io/version"]; ok {
+			return false
+		}
+		if _, ok := labels["release.giantswarm.io/version"]; ok {
+			return false
+		}
+		if _, ok := labels["cluster-operator.giantswarm.io/version"]; ok {
 			return false
 		}
 		return true
-	default:
-		return false
 	}
+
+	switch v := obj.(type) {
+	case *capiv1alpha3.Cluster:
+		return checker(v.Labels)
+	case *capiv1alpha4.Cluster:
+		return checker(v.Labels)
+	case *capiv1beta1.Cluster:
+		return checker(v.Labels)
+	}
```

Comment:
Why the switch? why not always to call `checker(v.Labels)` ?

Reply:
`.(type)` can only be used in a `switch`